blueflood-core/src/main/java/com/rackspacecloud/blueflood/threading/ThreadPoolSizedExecutorServiceAdapter.java

```diff
@@ -0,0 +1,104 @@
+package com.rackspacecloud.blueflood.threading;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.*;
+
+/**
+ * An adapter class so that we can use {@code ThreadPoolExecutor} with
+ * {@link SizedExecutorService}, to support dependency injection in the tests.
+ *
+ * TODO:
+ * {@link com.rackspacecloud.blueflood.service.RollupBatchWriter#enqueueRollupForWrite}
+ * makes calls to {@link #getActiveCount()} and {@link #getPoolSize()}, which
+ * means it must be hard-coded to use the {@link ThreadPoolExecutor} class
+ * instead of the {@link ExecutorService} interface. Once that method is fixed
+ * and no longer relies on those calls, it can be changed to use just
+ * {@link ExecutorService}, and both
+ * {@link ThreadPoolSizedExecutorServiceAdapter} and
+ * {@link SizedExecutorService} should be removed.
+ */
+public class ThreadPoolSizedExecutorServiceAdapter implements SizedExecutorService {
+
+    private final ThreadPoolExecutor executor;
+
+    public ThreadPoolSizedExecutorServiceAdapter(ThreadPoolExecutor executor) {
+        if (executor == null) throw new NullPointerException();
+        this.executor = executor;
```

Comment:
We should decide on a convention and run checkstyle, or something like that. We could run it in travis so that naming conventions are applied consistency and show up in PR's.